client/signup/jetpack-connect/authorize-form.jsx

```diff
@@ -640,24 +636,27 @@ const JetpackConnectAuthorizeForm = React.createClass( {
 
 export default connect(
 	state => {
-		const site = state.jetpackConnect.jetpackConnectAuthorize &&
-				state.jetpackConnect.jetpackConnectAuthorize.queryObject &&
-				state.jetpackConnect.jetpackConnectAuthorize.queryObject.site
+		const queryObjectSite = state.jetpackConnect.jetpackConnectAuthorize &&
+			state.jetpackConnect.jetpackConnectAuthorize.queryObject &&
+			state.jetpackConnect.jetpackConnectAuthorize.queryObject.site;
+
+		const site = queryObjectSite
 			? getSiteByUrl( state, state.jetpackConnect.jetpackConnectAuthorize.queryObject.site )
```

Comment:
>that queryObject is pretty much what Jetpack sends to calypso, so renaming it is out of reach (until a new jetpack release).

Ah, I didn't realize this. Well, another alternative to changing the structure of state is to use more selectors, so in this case `getSiteByUrl( state, getJetpackConnectAuthorizeSiteId( state ) );` which goes to the point of both making it more obvious that it's a site ID and also making state "opaque" (i.e. allowing us to change the structure later without too much trouble if we decide to).